### PR TITLE
Prime Numbers in Python Article

### LINKED ISSUE #861: Add Article for Prime Number in Python
See #860

### PR DESCRIPTION
**Pull request by hmajid2301**
Added prime number article in Python, also updated
the `prime-number.py` it wouldn't detect 2 as prime
 but would detect 1 as prime.

Fixes #861

Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [x] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #861 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: docs/projects/prime-number/_posts/2019-01-15-python.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+The `not bool(...)` then reverses this value, if `bool("a") = True` then `not bool("a") = False`.
+Then we return this value `return not bool(...)`. 
+
+Using our `x = 17` example again `[n for n in range(3, int(ceil(sqrt(x))+1)) if x % n == 0]` = `[]`.
+So `return not bool([])` = `return not False` = `return True`, so 17 is prime.
+
+The list that gets generated stores all factors of x except 1 and `17`. So if the list is empty `x` must be prime.
+If it contains any factors then it cannot be prime.
+
+`https://www.pexels.com/photo/grey-3-chain-1061140/`
```
Comment (hmajid2301) [inline docs/projects/prime-number/_posts/2019-01-15-python.md]: need to remove this

**Comment by hmajid2301** [comment]
Image from https://www.pexels.com/photo/grey-3-chain-1061140/

File: archive/p/python/prime-number.py [hunk 2]
```diff
@@ -4,9 +4,7 @@
 
 
 def is_prime(x):
-    if x <= 1:
-        return True
-    if x % 2 == 0:
+    if (x % 2 == 0 and x is not 2) or (x == 1):
```
Comment (jrg94) [inline archive/p/python/prime-number.py]: It appears these changes break the solution:

```
0 -> Usage: please enter non-negative value
1 -> False
2 -> True
3 -> True
4 -> False
```

Comment (jrg94) [reply archive/p/python/prime-number.py]: Nevermind! My brain went blank here. This totally works.

Comment (jrg94) [reply archive/p/python/prime-number.py]: That said, we exclude 0 which doesn't match the usage statement.

Comment (hmajid2301) [reply archive/p/python/prime-number.py]: ahh ok i'll fix the 0.

**Comment by jrg94** [comment]
Actually, let’s talk about 0. Is it prime or not? I want to document that either way. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by hmajid2301** [comment]
> Actually, let’s talk about 0. Is it prime or not? I want to document that either way.
> 
> Sent with [GitHawk](http://githawk.com)

I don't think it's prime https://math.stackexchange.com/questions/539174/is-zero-a-prime-number. Since it's even it cannot be a prime by definition.

**Comment by jrg94** [comment]
Great! I’m going to update #953, so we can nail that down. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jrg94** [comment]
See #954 for the new documentation for this problem. I'll take a look at your article shortly. Thanks for the help! 😄

File: docs/projects/prime-number/_posts/2019-01-17-python.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+title: Prime Number in Python
+layout: default
+last-modified: 2019-01-17
+featured-image: prime-numbers-in-python-featured-image.JPEG
+tags: [python, prime-number]
+authors:
+  - hmajid2301
+---
+
+In this article, we'll be tackling checking if a number is a prime in Python.
```
[the remark this answers is not in the document]
Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]: Agreed! Good catch.

File: docs/projects/prime-number/_posts/2019-01-17-python.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+If the integer is less than 0 it's not a valid input so we return an error by calling the
+`exit_with_error()` function.
+
+We then call the main part of our program `print(is_prime(x))` which will check if the integer
+is prime and print this out on the terminal for the user
+
+Finally we wrap this entire block in a `try ... except`, we catch two exceptions `IndexError`
+will be thrown is `args` isn't a list and we try to access `args[0]`. `ValueError` will be thrown
+if you try to convert a non-integer string into an integer for example if args[0] was `"a"` ->
+`int("a")`. If any exceptions are raised then we call the `exit_with_error()` function.
```
[the remark this answers is not in the document]
Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]: I would even go as far as to list the exceptions at the end of the first sentence.

Finally we wrap this entire block in a try ... except, and we catch two exceptions: `IndexError` and `ValueError`.

File: docs/projects/prime-number/_posts/2019-01-17-python.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+The `not bool(...)` then reverses this value, if `bool("a") = True` then `not bool("a") = False`.
+Then we return this value `return not bool(...)`. 
+
+Using our `x = 17` example again `[n for n in range(3, int(ceil(sqrt(x))+1)) if x % n == 0]` = `[]`.
+So `return not bool([])` = `return not False` = `return True`, so 17 is prime.
+
+The list that gets generated stores all factors of x except 1 and `17`. So if the list is empty `x` must be prime.
+If it contains any factors then it cannot be prime.
+
+`https://www.pexels.com/photo/grey-3-chain-1061140/`
```
Comment (jrg94) [inline docs/projects/prime-number/_posts/2019-01-17-python.md]: Can you remove this line? I'm not sure why it's here...

Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]:
```suggestion
```

File: docs/projects/prime-number/_posts/2019-01-17-python.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+### Prime Numbers
+
+```python
+def is_prime(x):
+    if (x % 2 == 0 and x is not 2) or (x == 1):
+        return False
+    return not bool([n for n in range(3, int(ceil(sqrt(x))+1)) if x % n == 0])
+```
+
+Now onto the main part of the program, this is the function that actually checks if the integer is
```
[the remark this answers is not in the document]
Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]:
```suggestion
Now onto the main part of the program. This is the function that actually checks if the integer is
```

File: docs/projects/prime-number/_posts/2019-01-17-python.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+### The Main Function
+
+Breaking down this solution bottom up,
+
+```python
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+This bit of code checks to see if this is the main module run. If it is then it calls the main
```
[the remark this answers is not in the document]
Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]:
```suggestion
This bit of code checks to see if this is the main module run. If it is, then it calls the main
```

File: docs/projects/prime-number/_posts/2019-01-17-python.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrg94) [reply docs/projects/prime-number/_posts/2019-01-17-python.md]:
```suggestion
In this article, we'll be tackling prime number checking in Python.
```

**Comment by hmajid2301** [comment]
I have applied the changes :)

**Comment by jrg94** [comment]
It looks like your changes haven't been pushed yet.

**Pull request merged**